pokemon-colorscripts: init at 2024-10-19-8fab453

viitorags opened the pull request:
This PR adds the new package `pokemon-colorscripts`.

It's a collection of scripts that display Pokémon sprites with colors in the terminal.
Homepage: https://gitlab.com/phoneybadger/pokemon-colorscripts

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [ ] NixOS test(s)
  - and/or [ ] package tests
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - [ ] made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- Nixpkgs 25.11 Release Notes (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- NixOS 25.11 Release Notes (https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

viitorags committed:
pokemon-colorscripts: init at 2024-10-19-8fab453

viitorags committed:
pokemon-colorscripts: Fix nixpkgs-vet issues (e.g., maintainers)

donovanglover commented:
Welcome to nixpkgs! Note that [`pokemon-colorscripts-mac`](https://github.com/nuke-dash/pokemon-colorscripts-mac) which we already package was forked from this in 2021 but seems inactive.

Personally I'd be in favor of only including [`krabby`](https://github.com/yannjor/krabby) in nixpkgs since it's a rewrite of `pokemon-colorscripts` but with more features.

viitorags committed:
pokemon-colorscripts: Apply review suggestions

 - Use SRI hash for src.
- Update version string format to "0-unstable-YYYY-MM-DD".
- Change substituteInPlace to use --replace-fail.
- Remove explicit declaration from pkgs/top-level/all-packages.nix.
(Note: installPhase mkdir -p retained as per testing)

Sigmanificient commented on pkgs/by-name/po/pokemon-colorscripts/package.nix:
```diff
@@ -0,0 +1,39 @@
+    substituteInPlace install.sh --replace /usr/local $out
+  '';
+
+  installPhase = ''
+    mkdir -p $out/opt
+    mkdir -p $out/bin
+    ./install.sh
+  '';
```
Whoops! Well can you add the `runHook`?

viitorags replied:
Thanks a lot, after taking a closer look, I realized I was making a mistake. I managed to use what you recommended, and it worked!

viitorags committed:
maintainers: Add viitorags

viitorags committed:
pokemon-colorscripts: fix installPhase by using runHook

viitorags committed:
maintainers: Apply treefmt formatting for viitorags

viitorags committed:
maintainers: Update viitorags entry per CI feedback (remove email)

viitorags committed:
pokemon-colorscripts: Update to commit more recent

viitorags committed:
Merge branch 'NixOS:master' into adding-pokemon-colorscripts

viitorags committed:
Merge branch 'NixOS:master' into adding-pokemon-colorscripts
